### PR TITLE
Add GH Actions workflow for integration testing with Sage

### PR DESCRIPTION
**Pull request by mkoeppe**

### CODE REVIEW

**Comment by embray** [comment]
This is quite a tour de force.  Have you used this for checking other packages' Sage integration?

**Comment by embray** [comment]
This runs so many jobs, and a lot of them are failing.  How do I know which failures matter and which ones don't (or do they?)

**Comment by embray** [comment]
I haven't really kept up with what you've been doing with tox, but it's pretty mind-bending.  I never would have even thought of using tox to run tests in Docker, much less have it do things like install Cygwin packages.  Does tox even create Python virtualenvs in these cases?  It seems like there would be no reason for it to but I'm still pretty confused about how this works.

File: .github/workflows/ci-sage.yml [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+  # Sage distribution packages to build
+  TARGETS_PRE: build/make/Makefile
+  TARGETS:     SAGE_CHECK=no SAGE_CHECK_PACKAGES=cysignals cysignals
+  TARGETS_OPTIONAL: build/make/Makefile
+  # Standard setting: Test the current beta release of Sage:
+  SAGE_REPO:   sagemath/sage
+  SAGE_REF:    develop
+  # Temporarily test with the branch from sage ticket
+  # (this is a no-op after that ticket is merged)
+  SAGE_TRAC_GIT: https://github.com/sagemath/sagetrac-mirror.git
```
Comment (embray) [inline .github/workflows/ci-sage.yml:62]: What exactly does this mean?  Would it be removed later?
Something I don't understand is that if this is meant to check Sage with a different version of cysignals, why would it need to reference a ticket related to upgrading cysignals?

**Comment by mkoeppe** [comment]
> This is quite a tour de force. Have you used this for checking other packages' Sage integration?

Yes, a version of this has been in use for a while now with Normaliz, arb, e-antic; and most recently a PR adding it to Singular was accepted.

**Comment by mkoeppe** [comment]
> Does tox even create Python virtualenvs in these cases? It seems like there would be no reason for it to 
It does, but the venvs are basically empty.

**Comment by mkoeppe** [comment]
> This runs so many jobs, 

Currently half of them are for macOS, where we are testing an excessive amount of variants because of the challenging Big Sur situation.

> and a lot of them are failing. How do I know which failures matter and which ones don't (or do they?)

This is a known weakness of this approach of sending Sage's portability infrastructure to upstream packages: Unrelated issues in dependencies of the tested package and instabilities in the distributions can lead to failures, which may be hard to understand, or just annoying, for the maintainer of the upstream package.

The failure for `cygwin-minimal` looks real to me:
```
src/cysignals/tests.pyx
killed by signal: 11
make[3]: *** [Makefile:71: check-doctest] Error 1
```
Also `ubuntu-trusty-minimal` looks like a real error in `cysignals`, but I don't see an error message.

I haven't looked at the other failing runs

**Comment by mkoeppe** [comment]
ping?

**Comment by embray** [comment]
I wonder if we could reduce the number of variants to ones that just already work, just for starters (so I'm merging something that's working into master), than open a separate PR (or PRs) for adding other variants once they're working.

**Comment by mkoeppe** [comment]
That's fine with me, please go ahead - certainly green checkmarks are more reassuring to look at

**Comment by mkoeppe** [comment]
Do you want this on master or on 1.10.x?

**Comment by mkoeppe** [comment]
@kliem  Running at https://github.com/mkoeppe/cysignals/actions/runs/1283405525

**Comment by kliem** [comment]
This seems to work and I would propose on merging this, so that we have something to test pull requests with.

**Comment by mkoeppe** [comment]
Sounds good to me

**Comment by kliem** [comment]
So for merging we comment out lines 62--64 and leave them there so we can use it, once we need it.

**Comment by mkoeppe** [comment]
> So for merging we comment out lines 62--64 and leave them there so we can use it, once we need it.

I think it also works as is. When a change in Sage is needed, it will suffice to just put it on the branch of the target ticket https://trac.sagemath.org/ticket/32579

**Comment by kliem** [comment]
I see.

**Comment by kliem** [comment]
This workflow has already proven itself very useful at #139. Thank you.

**Pull request merged**